src/main/java/com/lasterbergamot/balldontlie/domain/game/service/impl/GameServiceImpl.java

```diff
@@ -67,19 +73,67 @@ public void getAllGamesFromBalldontlieAPI() {
     }
 
     @Override
-    public List<Game> getGames(final int count) {
-        List<Game> games = gameRepository.findAll();
-
-        return count == -1
-                ? games
-                : games.stream().limit(count).collect(Collectors.toUnmodifiableList());
+    public List<Game> getGames(final Optional<Integer> count) {
+        return gameRepository
+                .findAll()
+                .stream()
+                .limit(count.orElse(Integer.MAX_VALUE))
+                .collect(Collectors.toUnmodifiableList());
     }
 
     @Override
     public Optional<Game> getGame(final int id) {
         return gameRepository.findById(id);
     }
 
+    @Override
+    public Game createGame(String date,
+                           Integer homeTeamScore, Integer visitorTeamScore,
+                           Integer season, Integer period,
+                           String status, String time,
+                           Boolean postseason, Integer homeTeamId,
+                           Integer visitorTeamId) {
+        Map<String, Team> validationResults = validateMutationInputs(homeTeamId, visitorTeamId);
+
+        AtomicInteger atomicGameId = new AtomicInteger(gameRepository.getMaximumId());
```

Comment:
Are you using AtomicInteger here because of concurrency issues? You are right that it can happen, but unfortunately AtomicInteger won't help us here :( You can just use plain int.
But I don't know how to guarantee a unique ID in a concurrent environment, let me check if i can find something...

Reply:
Done! 🔥